components/uniset/src/conversions.rs

```diff
@@ -0,0 +1,123 @@
+use crate::utils::deconstruct_range;
+use crate::UnicodeSet;
+use std::{
+    convert::TryFrom,
+    ops::{Range, RangeBounds, RangeFrom, RangeFull, RangeInclusive, RangeTo, RangeToInclusive},
+};
+
+fn try_from_range_impl(range: impl RangeBounds<char>) -> Result<UnicodeSet, (u32, u32)> {
+    let (from, till) = deconstruct_range(range);
+    if from < till {
+        let set = vec![from, till];
+        Ok(UnicodeSet::try_from(set).unwrap())
+    } else {
+        Err((from, till))
+    }
+}
+
+impl TryFrom<Range<char>> for UnicodeSet {
+    type Error = String;
+
+    fn try_from(range: Range<char>) -> Result<Self, Self::Error> {
+        match try_from_range_impl(range) {
+            Ok(u) => Ok(u),
+            Err((from, till)) => Err(format!("Range must be ascending: {} - {}", from, till)),
```

Comment:
Please don't use unconditional `format!` -- it bloats code size big time.  Better style is to make an enum variant of an Error enum for your component, and then you can write the message in the `Debug` trait of the Error enum.

Reply:
agreed and changed as below